src/dict.rs

```diff
@@ -0,0 +1,97 @@
+use std::{collections::HashMap, hash::{DefaultHasher, Hash, Hasher}};
+
+use crate::{alloc::Alloc, str::Str, vm::Value};
+
+#[derive(Clone)]
```

Comment:
I don't think that we should derive clone, probably want to implement our own clone like with arrays?

Cloning needs to happen in deepcopy.rs, which is used for messaging and GC.

Reply:
good spotting, I had missed it entirely while copying the original impl